### PR TITLE
Add guideline for including a sample in sample test

### PR DESCRIPTION
**Pull request by numerology**
Part of #1750 
Add a section in README.md to help contributor streamline the effort of authoring new samples.

---
This change is [Reviewable](https://reviewable.io/reviews/kubeflow/pipelines/2026)

### CODE REVIEW

File: samples/README.md [hunk 1]
```diff
@@ -33,3 +33,25 @@ Useful parameter values:
...
+
+## (Optional) Add sample test coverage
+
+For those samples that cover critical functions of KFP, possibly it should be picked up by KFP's sample test
+so that it won't be broken by accidental PR merge. To do that, the sample to be covered should meet the following 
+requirements/conventions.
+
+* Under the core sample directory `kubeflow/pipelines/samples/core`
+* Make sure that the sample is either `*.py` or `*.ipynb`, and its file name is in consistence with its dir name.
+* For `*.py` sample, its main invokes `kfp.compiler.Compiler().compile()` to compile the pipeline function into pipeline
```
Comment (gaoning777) [inline samples/README.md]: This is the recommendation for samples, not required for sample test, isn't it?

File: samples/README.md [hunk 2]
```diff
@@ -33,3 +33,25 @@ Useful parameter values:
 ## Components source
 
 All samples use pre-built components. The command to run for each container is built into the pipeline file.
+
+## (Optional) Add sample test coverage
```
Comment (gaoning777) [inline samples/README.md]: Some of the following are conventions required by the sample tests and some are steps to take in order to add the sample test. Would you mind separating these two. For example, ''sample conventions" and "add sample test"

File: samples/README.md [hunk 3]
```diff
@@ -33,3 +33,25 @@ Useful parameter values:
...
+* Under the core sample directory `kubeflow/pipelines/samples/core`
+* Make sure that the sample is either `*.py` or `*.ipynb`, and its file name is in consistence with its dir name.
+* For `*.py` sample, its main invokes `kfp.compiler.Compiler().compile()` to compile the pipeline function into pipeline
+yaml spec.
+* For `*.ipynb` sample, parameters (e.g., experiment name and project name) should be defined in a dedicated cell and 
+tagged as parameter. Detailed guideline is [here](https://github.com/nteract/papermill). Also, custom environment setup
+should be done within the notebook itself by `!pip install`
+* If pipeline running requires additional pipeline parameters, they can be specified in a config yaml file
+placed under `test/sample-test/configs`. See 
+`test/sample-test/configs/tfx_cab_classification.config.yaml` as an example. The config yaml file will be validated
```
Comment (gaoning777) [inline samples/README.md]: Maybe add the link?

File: samples/README.md (hunk 1 above)
Comment (numerology) [reply samples/README.md]: Wait. I think currently sample test needs to submit a pipeline run. Did I miss something?

Comment (gaoning777) [reply samples/README.md]: But this is not a hard requirement for the samples. The sample test infra contains the compilation step.

Comment (numerology) [reply samples/README.md]: Yes got it.

File: samples/README.md (hunk 2 above)
Comment (numerology) [reply samples/README.md]: Done.

File: samples/README.md (hunk 3 above)
Comment (numerology) [reply samples/README.md]: Done.

File: samples/README.md [hunk 4]
```diff
@@ -33,3 +33,30 @@ Useful parameter values:
...
+tagged as parameter. Detailed guideline is [here](https://github.com/nteract/papermill). Also, custom environment setup
+should be done within the notebook itself by `!pip install`
+
+
+## (Optional) Add sample test
+For those samples that cover critical functions of KFP, possibly it should be picked up by KFP's sample test
+so that it won't be broken by accidental PR merge. Contributors can do that by following these steps.
+
+* Place the sample under the core sample directory `kubeflow/pipelines/samples/core`
+* Make sure it follows the sample conventions.
```
Comment (gaoning777) [inline samples/README.md]: Add the link to jump to the 'sample convention' section?

Comment (numerology) [reply samples/README.md]: Done.

Comment (gaoning777) [reply samples/README.md]: Why not the relative link: https://support.typora.io/Links/?
markdown supports that.

Comment (numerology) [reply samples/README.md]: Thanks. Just learnt :p

**Pull request merged**